### PR TITLE
oc_reset_v1: allow to set close_all_tls_connections_immediately

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

**Comment by jkralik** [comment]
~format

**Comment by Danielius1922** [comment]
@jkralik what about a unit test for the new functions? You should be able to mock a connection using `oc_tls_add_peer` and then the reset should remove it.

File: messaging/coap/engine.c [hunk 1]
```diff
@@ -924,14 +910,6 @@ coap_receive(oc_message_t *msg)
     }
   }
 
-#ifdef OC_SECURITY
-  if (coap_status_code == CLOSE_ALL_TLS_SESSIONS) {
```
Comment (Danielius1922) [inline messaging/coap/engine.c:928]: This was redundant?

Comment (jkralik) [reply messaging/coap/engine.c:928]: Yes. it was not obvious when the tls connections will be closed. - refactored

**Comment by jkralik** [comment]
~format

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**